Documentation/2.1-update-process.md

```diff
@@ -0,0 +1,62 @@
+* src - Contains a submodule for each repo.
+
+## Build Stages
+
+The source-build process consists of 3 main stages:
+
+1. **Build source-build repo**
+
+    * Install tools (CLI and BuildTools)
+    * Apply temporary patches to repos as applicable
+    * Build each repo
+
+1. **Generate tarball**
+    * Copy Tools to tarball
+    * Copy sources to tarball
+    * Create prebuilt packages directory
+        * Copy each restored package to the prebuilt directory
+        * Remove any packages generated as part of the source-build build
+
+1. **Build tarball in an offline environment**
+
+## Auto-dependency flow API
+
+Source-build defines a set of properties to assist in building and placing output.  These are:
+
+* `DotNetPackageVersionPropsPath` - To identify the versions of built pacakges to depend on.
+* `DotNetBuildOffline` - Identifies that build is indended to be offline.
+* `DotNetRestoreSourcePropsPath` - Where to restore packages from during build.
+* `DotNetOutputBlobFeedDir` - Where to place output during build.
+
+These are defined in [Auto-dependency flow API](https://github.com/dotnet/source-build/blob/dev/release/2.1/Documentation/auto-dependency-flow/api.md)
+
+## Updating Source-build for 2.1
+
+### 2.1-Specific Source-build Goals
+
+Source-build has been updated to the commit SHAs identified in the 2.1-preview1 prodcon build.  The goal is to get source-build to build correctly with these SHAs.  Once that is complete, we can enable auto-update of commit SHAs from each successful prodcon build and ensure that source-build doesn't get out of sync.  To take a measured approach to onboarding repos, a known good project has been defined which contains the repos that can successfully build in source-build.  The goal with the 2.1 exercise is to add repos to the known-good project and get them to build correctly.
+
+### Repo Owners Responsibilities
+
+Within these three stages, these are the things for repo owners to validate to ensure their repos are buildng correctly.
+
+* Make sure the 2.1-preview1 commit SHA is correct for the repo submodule
+* Validate the build command used to build the repo (see source-build/repos/\<reponame>.proj)
+* Incorporate logic from patches into the repo code base to eliminate patches in source-build
+* Ensure that packages restored during stage 1 build are archived at either:
+  * `$NUGET_PACKAGES` (bin/obj/x64/Release/nuget-packages/\<reponame>) or
+  * \<root>/packages
+* Ensure built packages end up in `$(DotNetOutputBlobFeedDir)packages/`
+* Ensure that dependency versions for repo dependencies are picked up from `$(DotNetPackageVersionPropsPath)`
+* Ensure that all other dependent packages are picked up from `$(DotNetRestoreSourcePropsPath)`
```

Comment:
should call out they are only picked up from here and not other places in offline mode. Which means repo's need to correctly condition their builds to only pull packages from here.

Reply:
Updated.